### PR TITLE
 Update the Spot hibernate agent to do nothing if ODH is configured

### PR DESCRIPTION
**Pull request by achojeon**
*Issue #, if available:*

*Description of changes:*
Update the Spot hibernate agent to do nothing if ODH is configured
to avoid any conflicts between two agents

*Testing 
Modified Spot hibernate agent on two different spot instances that have hibernate-option.configured true or false to test the change. Successfully verify the hibernate-option.configured meta-data of each instances

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/hibagent [hunk 1]
```diff
@@ -39,6 +39,9 @@ SWAP_RESERVED_SIZE = 16384
 log_to_syslog = True
```
Comment (ee-gitbub) [inline agent/hibagent:40]: Can you please add to the commit message a description of how you tested it?

File: agent/hibagent [hunk 2]
```diff
@@ -497,6 +500,36 @@ def adjust_pm_timeout(timeout):
         log("Failed to adjust pm_freeze_timeout to %d. Error: %s" % (timeout, str(e)))
         exit(1)
 
+def get_imds_token(seconds=21600):
+    """ Get a token to access instance metadata. """
+    log("Requesting new IMDSv2 token.")
+    request_header = {'X-aws-ec2-metadata-token-ttl-seconds': '{}'.format(seconds)}
```
Comment (ee-gitbub) [inline agent/hibagent:507]: Should we specify retry value:
TOKEN=$(curl --silent --fail --retry 3 -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")

File: agent/hibagent [hunk 3]
```diff
@@ -497,6 +500,36 @@ def adjust_pm_timeout(timeout):
...
+    token_url = '{}/{}'.format(IMDS_BASEURL, IMDS_API_TOKEN_PATH)
+    response = requests.put(token_url, headers=request_header)
+    response.close()
+    if response.status_code != 200:
```
Comment (ee-gitbub) [inline agent/hibagent]: response.ok (instead of 200)

File: agent/hibagent [hunk 4]
```diff
@@ -497,6 +500,36 @@ def adjust_pm_timeout(timeout):
...
+    imds_token = get_imds_token()
+    if imds_token is None:
+        # IMDS http endpoint is disabled
+        return False
+
+    request_header = {'X-aws-ec2-metadata-token': imds_token}
+    response = requests.get("{}/{}".format(IMDS_BASEURL, IMDS_SPOT_ACTION_PATH),
+                 headers=request_header)
+    response.close()
+    if response.status_code != 200 or response.text.lower() == "false":
```
Comment (ee-gitbub) [inline agent/hibagent]: response.ok (instead of 200)

File: agent/hibagent (hunk 2 above)
Comment (ee-gitbub) [reply agent/hibagent:507]: Here is a Python code implementation for  retries over request.put() :
https://code.amazon.com/packages/AdaptDHAL/blobs/ea7890aad289411b5badfa77d48a05cbda95eedd/--/src/adapt_dhal/devices/features/packages/dlipowerswitch/dlipowerswitch.py#L66

**Review by einynot — APPROVED** [review]

File: agent/hibagent (hunk 4 above)
Comment (achojeon) [reply agent/hibagent]: Modified!

File: agent/hibagent (hunk 3 above)
Comment (achojeon) [reply agent/hibagent]: Modified !

File: agent/hibagent (hunk 2 above)
Comment (achojeon) [reply agent/hibagent:507]: We discussed and concluded the retry policy is not necessary.

File: agent/hibagent (hunk 1 above)
Comment (achojeon) [reply agent/hibagent:40]: Added !

File: agent/hibagent [hunk 5]
```diff
@@ -497,6 +501,36 @@ def adjust_pm_timeout(timeout):
...
+        return None
+
+    return response.text
+
+def hibernation_enabled():
+    """Returns a boolean indicating whether hibernation-option.configured is enabled or not."""
+
+    imds_token = get_imds_token()
+    if imds_token is None:
+        # IMDS http endpoint is disabled
```
Comment (eptiger) [inline agent/hibagent:522]: Should we log a message in this case?

Comment (achojeon) [reply agent/hibagent:522]: Modified !

**Pull request merged**